devito/ir/equations/algorithms.py

```diff
@@ -213,18 +213,24 @@ def _(expr, mapper, rebuilt, sregistry):
     _concretize_subdims(expr.implicit_dims, mapper, rebuilt, sregistry)
 
 
+@_concretize_subdims.register(Thickness)
+def _(tkn, mapper, rebuilt, sregistry):
+    if tkn in mapper:
+        # Already have a substitution for this thickness
+        return
+
+    mapper[tkn] = tkn._rebuild(name=sregistry.make_name(prefix=tkn.name))
+
+
 @_concretize_subdims.register(SubDimension)
 def _(d, mapper, rebuilt, sregistry):
     if d in mapper:
         # Already have a substitution for this dimension
         return
 
-    tkns_subs = {tkn: tkn._rebuild(name=sregistry.make_name(prefix=tkn.name))
-                 for tkn in d.tkns}
-    left, right = [mM.subs(tkns_subs) for mM in (d.symbolic_min, d.symbolic_max)]
-    thickness = tuple((v, d._thickness_map[k]) for k, v in tkns_subs.items())
-
-    mapper[d] = d._rebuild(symbolic_min=left, symbolic_max=right, thickness=thickness)
+    tkns = tuple(t._rebuild(name=sregistry.make_name(prefix=t.name)) for t in d.tkns)
+    mapper.update({t0: t1 for t0, t1 in zip(d.tkns, tkns)})
```

Comment:
I would avoid t0 and t1 due to name clash with time buffers.
It's not about correctness, but mostly when grepping